osv/osv.go

```diff
@@ -0,0 +1,165 @@
+package osv
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/aquasecurity/vuln-list-update/utils"
+	"github.com/cheggaaa/pb"
+	"github.com/spf13/afero"
+	"golang.org/x/xerrors"
+)
+
+const (
+	securityTrackerURL = "https://osv-vulnerabilities.storage.googleapis.com/%s/all.zip"
+	osvDir             = "osv"
+)
+
+var defaultEcosystemDirs = map[string]string{
+	"PyPI":      "python",
+	"Go":        "go",
+	"crates.io": "rust",
+}
+
+type options struct {
+	url           string
+	dir           string
+	ecosystemDirs map[string]string
+}
+
+type option func(*options)
+
+type Osv struct {
+	opts *options
+}
+
+func WithURL(url string) option {
+	return func(opts *options) {
+		opts.url = url
+	}
+}
+
+func WithDir(dir string) option {
+	return func(opts *options) {
+		opts.dir = dir
+	}
+}
+
+func WithEcosystem(ecosystemDir map[string]string) option {
+	return func(opts *options) {
+		opts.ecosystemDirs = ecosystemDir
+	}
+}
+
+func NewOsv(opts ...option) Osv {
+	o := &options{}
+
+	for _, opt := range opts {
+		opt(o)
+	}
+
+	if len(o.ecosystemDirs) == 0 {
+		ecosystemDirs := make(map[string]string)
+		for name, dir := range defaultEcosystemDirs {
+			ecosystemDirs[name] = dir
+		}
+		o.ecosystemDirs = ecosystemDirs
+	}
+
+	if o.url == "" {
+		o.url = securityTrackerURL
+	}
+
+	if o.dir == "" {
+		o.dir = filepath.Join(utils.VulnListDir(), osvDir)
+	}
+
+	return Osv{
+		opts: o,
+	}
+}
+
+func (osv *Osv) Update() error {
+	allfiles, err := osv.getAllFiles()
+	if err != nil {
+		return xerrors.Errorf("failed to get files: %w", err)
+	}
+
+	bar := pb.StartNew(osv.getAmountFiles(allfiles))
```

Comment:
I think it is fast enough. No need to show a progress bar.

Reply:
Removed